alacritty/src/window_context.rs

```diff
@@ -470,30 +481,12 @@ impl WindowContext {
             self.mouse.hint_highlight_dirty = false;
         }
 
-        // Skip rendering until we get a new frame.
-        if !self.display.window.has_frame.load(Ordering::Relaxed) {
-            return;
-        }
-
-        if self.dirty && !self.occluded {
-            // Force the display to process any pending display update.
-            self.display.process_renderer_update();
-
-            self.dirty = false;
-
-            // Request immediate re-draw if visual bell animation is not finished yet.
-            if !self.display.visual_bell.completed() {
-                self.display.window.request_redraw();
-            }
-
-            // Redraw the window.
-            self.display.draw(
-                terminal,
-                scheduler,
-                &self.message_buffer,
-                &self.config,
-                &self.search_state,
-            );
+        // Request a redraw in case we need to draw a frame and we haven't asked for it yet..
+        //
+        // Even though redraw requests are squashed on winit side, we try not to request
+        // more if we haven't received the one we handled before.
+        if self.dirty && !self.occluded && !matches!(event, WinitEvent::RedrawRequested(_)) {
```

Comment:
Not requesting a redraw when occluded means we have to handle this when occlusion ends, right?

I don't see this being handled in the occlusion event.

Reply:
The `Occlusion` change will result in generating an event, meaning that it'll change to `false` thus we'll request a frame.

Reply:
Good point, I didn't notice this was after the event handling code.